tests/end2end/test_partition.py

```diff
@@ -204,3 +205,116 @@ def test_part_crud(dpm_mode_cpcs):  # noqa: F811
 
         with pytest.raises(zhmcclient.NotFound):
             cpc.partitions.find(name=part_name_new)
+
+
+def test_console_list_permitted_partitions(dpm_mode_cpcs):  # noqa: F811
+    # pylint: disable=redefined-outer-name
+    """
+    Test list permitted partitions on a cpc
+    """
+    if not dpm_mode_cpcs:
+        pytest.skip("HMC definition does not include any CPCs in DPM mode")
+
+    for cpc in dpm_mode_cpcs:
+        assert cpc.dpm_enabled
+
+        session = cpc.manager.session
+        hd = session.hmc_definition
+
+        client = zhmcclient.Client(session)
+        console = client.consoles.console
+
+        part_list = cpc.partitions.list()
+        if not part_list:
+            skip_warn("No partitions on CPC {c} managed by HMC {h} for "
+                      "the user {u}".format(c=cpc.name, h=hd.host,
+                                            u=session.userid))
+        part_list = pick_test_resources(part_list)
+
+        user_name = role_name = TEST_PREFIX + '_test_list_permitted_partitions'
```

Comment:
This is a test about listing partitions. There is no need to create or delete any users or password rules.

Reply:
As discussed, I thought of testing hmc functionality, Now I'm testing the list_permitted_partitions functionality